### PR TITLE
Add types to the `append()` methods of the `Table` and `InlineTable` classes

### LINKED ISSUE #281: Missing types in the `append()` method of the `Table` class
Hi! 👋 

I think the types for the [`append()` method of the `Table` class](https://github.com/sdispater/tomlkit/blob/master/tomlkit/items.py#L1577) are missing, as this method is part of the "public API". If adding them makes sense and I can help, let me know. 

Thanks! 😄

### PR DESCRIPTION
**Pull request by joaopalmeiro**
Closes #281

### CODE REVIEW

**Comment by frostming** [comment]
Do you mind helping fix the lint error?

**Comment by joaopalmeiro** [comment]
> Do you mind helping fix the lint error?

Sure!

File: tomlkit/items.py [hunk 1]
```diff
@@ -1574,7 +1574,7 @@ def __copy__(self) -> "Table":
             self.display_name,
         )
 
-    def append(self, key, _item):
+    def append(self, key: Union[Key, str], _item: Any) -> "Table":
```
Comment (frostming) [inline tomlkit/items.py:1577]: It seems key can be `None`, just add `None` to the Union

Comment (joaopalmeiro) [reply tomlkit/items.py:1577]: Sure! So, these methods will have the same types as the `append()` method of the `Container` class, right?

Comment (joaopalmeiro) [reply tomlkit/items.py]: @frostming , added `None` to the `append()` and `raw_append()` methods and fixed the linting errors. Can you check the MR again, please?

**Review by frostming — APPROVED** [review]

**Pull request merged**